controllers/operands/kubevirt.go

```diff
@@ -162,13 +162,14 @@ var (
 
 // KubeVirt feature gates that are exposed in HCO API
 const (
-	kvDownwardMetrics        = "DownwardMetrics"
-	kvWithHostPassthroughCPU = "WithHostPassthroughCPU"
-	kvRoot                   = "Root"
-	kvDisableMDevConfig      = "DisableMDEVConfiguration"
-	kvPersistentReservation  = "PersistentReservation"
-	kvAutoResourceLimits     = "AutoResourceLimitsGate"
-	kvAlignCPUs              = "AlignCPUs"
+	kvDownwardMetrics           = "DownwardMetrics"
+	kvWithHostPassthroughCPU    = "WithHostPassthroughCPU"
+	kvRoot                      = "Root"
+	kvDisableMDevConfig         = "DisableMDEVConfiguration"
+	kvPersistentReservation     = "PersistentReservation"
+	kvAutoResourceLimits        = "AutoResourceLimitsGate"
+	kvAlignCPUs                 = "AlignCPUs"
+	kvDynamicPodInterfaceNaming = "DynamicPodInterfaceNaming"
```

Comment:
@EdDev FYI.

Reply:
> If HCO PrimaryUserDefinedNetworkBinding FG is enabled,
> enable Kubevirt DynamicPodInterfaceNaming FG that is required
> for dynamic pod interface name detection.

It looks like the FG is used as a configuration knob.
The FG represents the feature graduation stage and therefore it cannot have dependencies like presented here.

If HCO is enabling a FG, it means it has guarantees it will never be removed, which is hard to commit to for Alpha or Beta features.

In this specific case, the dynamic pod iface name detection is not assured to be available in future Kubevirt releases. It needs to graduate to GA on Kubevirt and only then be permanently added to HCO as enabled. And it should not be dependent on any other feature lifecycle.

Reply:
look here https://github.com/kubevirt/hyperconverged-cluster-operator/pull/3139/files#r1836074173